### PR TITLE
upgrade rivet and yoda and fix spec files

### PR DESCRIPTION
**Pull request by mkirsano**

### CODE REVIEW

File: rivet-2.2.1.patch [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
++      , _vetoScale{20.*GeV, 30.*GeV}
++      , _yFiducial{4.4, 2.4}
++      , _gapCategories{"inclusive","gap"}
+     {
+       // Cannot do this in initialiser list without C++11
+-      _fiducialRegions = boost::assign::list_of<unsigned int>(2010)(2011);
+-      _vetoScale = boost::assign::list_of<double>(20.*GeV)(30.*GeV);
+-      _yFiducial = boost::assign::list_of<double>(4.4)(2.4);
+-      _gapCategories = boost::assign::list_of<std::string>("inclusive")("gap");
++//      _fiducialRegions = boost::assign::list_of<unsigned int>(2010)(2011);
```
Comment (ghost) [inline rivet-2.2.1.patch]: Could you please change the patch to just remove unused code instead of comment it?

**Comment by ghost** [comment]
+1
Tested compilation and compilation of GeneratorInterface/RivetInterface against it.
This is a backport from 75X and since no one complained about that I'm merging this one.

**Comment by smuzaffar** [comment]
@davidlange6 and @mkirsano , this will be forward ported to 75X too (due to automatic forward port of  cmsdist/74x->cmsdist/75x) .... is that OK ?

**Comment by smuzaffar** [comment]
forget my last comment,  this is already a back port from 75X (changes are already in 75X) :-)

**Pull request merged**